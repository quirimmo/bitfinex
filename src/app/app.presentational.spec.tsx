import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import App from './app.presentational';
import { of } from 'rxjs';

let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
const mockStartLoading = jest.fn();
const mockStopLoading = jest.fn();
const mockFetchSymbols = jest.fn().mockReturnValue(of([]));
const mockFetchCurrencies = jest.fn().mockReturnValue(of([]));
const mockFetchTickers = jest.fn().mockReturnValue(of([]));

describe('App Presentational Component', () => {
	beforeEach(() => {
		jest.useFakeTimers();
		component = shallow(
			<App
				fetchSymbols={mockFetchSymbols}
				fetchCurrencies={mockFetchCurrencies}
				fetchTickers={mockFetchTickers}
				startLoading={mockStartLoading}
				stopLoading={mockStopLoading}
			/>
		);
	});
	afterEach(() => {
		jest.clearAllMocks();
	});

	it('should be defined', () => {
		expect(component).toBeDefined();
	});

	it('should match the snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	describe('componentDidMount', () => {
		it('should call the startLoading prop', () => {
			expect(mockStartLoading).toHaveBeenCalled();
		});

		it('should call the stopLoading prop', () => {
			expect(mockStopLoading).toHaveBeenCalled();
		});

		it('should call the fetchSymbols prop', () => {
			expect(mockFetchSymbols).toHaveBeenCalled();
		});

		it('should call the fetchCurrencies prop', () => {
			expect(mockFetchCurrencies).toHaveBeenCalled();
		});

		it('should call the fetchTickers prop', () => {
			expect(mockFetchTickers).toHaveBeenCalled();
		});
	});
});
