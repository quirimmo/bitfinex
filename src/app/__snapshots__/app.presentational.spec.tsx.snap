// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Presentational Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    currencies={Array []}
    fetchCurrencies={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Observable {
              "_isScalar": true,
              "_subscribe": [Function],
              "value": Array [],
            },
          },
        ],
      }
    }
    fetchSymbols={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Observable {
              "_isScalar": true,
              "_subscribe": [Function],
              "value": Array [],
            },
          },
        ],
      }
    }
    fetchTickers={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Observable {
              "_isScalar": true,
              "_subscribe": [Function],
              "value": Array [],
            },
          },
        ],
      }
    }
    startLoading={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    stopLoading={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    symbols={Array []}
    tickers={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="app-loader-wrapper"
        >
          <Connect(AppLoader) />
        </div>,
        <Container
          fluid={true}
          tag="div"
        >
          <AppRoutes />
        </Container>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(AppLoader) />,
          "className": "app-loader-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <AppRoutes />,
          "fluid": true,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="app-loader-wrapper"
          >
            <Connect(AppLoader) />
          </div>,
          <Container
            fluid={true}
            tag="div"
          >
            <AppRoutes />
          </Container>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(AppLoader) />,
            "className": "app-loader-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <AppRoutes />,
            "fluid": true,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
